packages/css/src/index.ts

```diff
@@ -283,23 +298,30 @@ export const css = (args: ThemeUIStyleObject = {}) => (
 
   // insert variant props before responsive styles, so they can be merged
   // we need to maintain order of the style props, so if a variant is place in the middle
-  // of other props, it will extends its props at that same location order.
+  // of other props, it will extend its props at that same location order.
   if (obj && obj['variant']) {
     for (const key in obj) {
       const x = obj[key as keyof typeof styles]
       if (key === 'variant') {
-        const val = typeof x === 'function' ? x(theme) : x;
-        const variant = get(theme, val as string);
-        result = { ...result, ...variant };
+        // We could start using the types here to ensure we cover all public API.
+        const variantName = x as VariantProperty.Variant
+
+        // TODO: Is this even documented?
+        const val = typeof x === 'function' ? x(theme) : x
```

Comment:
@hasparus was saying something about removing/changing this? I don't recall exactly

Reply:
No idea what was going on there :) 

@hasparus - can you pls check, I think there were some suggestions that were really not part of this PR but for v1?